sdkgen/nodejs-request/lib/index.js

```diff
@@ -0,0 +1,76 @@
+// TODO add options and fetch options
+const getCodegenOptions = require('postman-code-generators').getOptions,
+  processCollection = require('./util').processCollection;
+
+/**
+ * Generates sdk for nodejs-request
+
+ * @param {PostmanCollection} collection - Postman collection Instance
+ * @param {Object} options - postman-code-generators options
+ * @param {Function} callback - callback functio to return results (err, response)
+ * @returns {String} - sdk snippet for input collection
+ * TODO add indentation to entire snippet
+ */
+function generate (collection, options, callback) {
+  var snippet = '',
+    collectionVariables = collection.variables ? collection.variables.members : [],
+    // sdkname = collection.name.split(' ').join('_'),
+    sdkname = 'SDKNAME',
+    indent = options.indentType === 'Tab' ? '\t' : ' ';
+  indent = indent.repeat(options.indentCount);
+  if (options.ES6_enabled) {
+    snippet += 'const ';
+  }
+  else {
+    snippet += 'var ';
+  }
+  snippet += 'request = require(\'request\');\n';
+  snippet += `function ${sdkname}(environment = null) {\n\n`;
+  snippet += options.ES6_enabled ? 'const ' : 'var ';
+  snippet += 'self = this;\n\n';
+  snippet += indent + 'this.requests = {\n';
+  processCollection(collection, options, (err, collectionSnippet) => {
+    if (err) {
+      return callback(err, null);
+    }
+    snippet += collectionSnippet;
+  });
+  snippet += indent + '};\n\n';
```

Comment:
Same problem here. processCollection is an asynchronous function. 
How is this working?

Reply:
Fixed, had some unpushed components